gpu: Support GPU passthrough to LXD containers using Container Device Interface (CDI) (#13562)
canonical/lxd: doc/config_options.txt

<!-- config group device-gpu-mdev-device-conf end -->
<!-- config group device-gpu-mig-device-conf start -->
```{config:option} id device-gpu-mig-device-conf

> :shortdesc: "DRM card ID of the GPU device or Container Device Interface (CDI) identifier of the card."

Reviewer: This is too long for a short description - can you break it up? (For example, have "ID of the GPU device" as short description and then explain the different options in the body text.)

Reviewer 2: For what it's worth, we use "Fully-qualified CDI device name" to mean `{{vendor}}/{{class}}={{name}}`

Author: fixed

```

```{config:option} id device-gpu-physical-device-conf
:shortdesc: "ID of the GPU device"
:type: "string"

> The ID can either be the DRM card ID of the GPU device or a fully-qualified Container Device Interface (CDI) name.
> The use of a fully-qualified CDI name is only allowed for a device added to a container, not a VM.
> Here are some examples of fully-qualified CDI names:

Reviewer: ```suggestion
The ID can either be the DRM card ID of the GPU device (container or VM) or a fully-qualified Container Device Interface (CDI) name (container only).

Here are some examples of fully-qualified CDI names:
```

- `nvidia.com/gpu=gpu0` : this instructs LXD to operate a discrete GPU (dGPU) passthrough of brand NVIDIA with the first discovered GPU on your system. You can use the `nvidia-smi` tool on your host to know which identifier to use.

> - `nvidia.com/gpu=igpu0` : this instructs LXD to operate an integrated GPU (iGPU) passthrough of brand NVIDIA with the first discovered iGPU on your system. An iGPU won't be detected by the `nvidia-smi` tool, so we encourage the user to use the `nvidia-ctk` tool through `nvidia-ctk cdi generate` to list the available iGPU devices.

Author: fixed

> - `nvidia.com/gpu=gpu0` : this instructs LXD to operate a discrete GPU (dGPU) passthrough of brand NVIDIA with the first discovered GPU on your system. You can use the `nvidia-smi` tool on your host to know which identifier to use.
> - `nvidia.com/gpu=igpu0` : this instructs LXD to operate an integrated GPU (iGPU) passthrough of brand NVIDIA with the first discovered iGPU on your system. An iGPU won't be detected by the `nvidia-smi` tool, so we encourage the user to use the `nvidia-ctk` tool through `nvidia-ctk cdi generate` to list the available iGPU devices.
> - `nvidia.com/gpu=all` : this instructs LXD to pass all the host GPUs of brand NVIDIA through the container.

Reviewer: ```suggestion
- `nvidia.com/gpu=gpu0`: Instructs LXD to operate a discrete GPU (dGPU) passthrough of brand NVIDIA with the first discovered GPU on your system. You can use the `nvidia-smi` tool on your host to find out which identifier to use.
- `nvidia.com/gpu=igpu0`: Instructs LXD to operate an integrated GPU (iGPU) passthrough of brand NVIDIA with the first discovered iGPU on your system. An iGPU won't be detected by the `nvidia-smi` tool, so you should use the `nvidia-ctk` tool through `nvidia-ctk cdi generate` to list the available iGPU devices.
- `nvidia.com/gpu=all`: Instructs LXD to pass all the host GPUs of brand NVIDIA through to the container.
```

Author: fixed
